src/Core/Client/Adapter/Http.php

```diff
@@ -12,12 +12,13 @@
 use Solarium\Core\Client\Endpoint;
 use Solarium\Core\Client\Request;
 use Solarium\Core\Client\Response;
+use Solarium\Core\Configurable;
 use Solarium\Exception\HttpException;
 
 /**
  * Basic HTTP adapter using a stream.
  */
-class Http implements AdapterInterface, TimeoutAwareInterface
+class Http extends Configurable implements AdapterInterface, TimeoutAwareInterface
```

Comment:
personally i'm not a huge fan of re-introducing the configurable extension here: wouldn't it make more sense to simply use a setter?

Reply:
I yust see that Curl adapter is using the Configurable, and I think it would be very confusion, when this both adapters dooing the same thing in a different way.

Reply:
using that sentiment, no library would progress...

the ``ConfigurableInterface`` is somewhat of a god class for this library and though for backwards compatability it might not be possible for us to remove it completely, we should be able to make the decission _not_ to use it

Reply:
And what is you suggestion not, to finaly finish this pull-request?